### ISSUE TITLE
ArrayList - unable to set value for first element

### ISSUE THREAD

**Issue by arthur-apanovics** [silverstripe/silverstripe-framework#9790]
## Affected Version
4.6.1

## Description
[ArrayList::offsetSet](https://github.com/silverstripe/silverstripe-framework/blob/a25e82fc14947a6d97bedd8a7354adfff2289a80/src/ORM/ArrayList.php#L847) uses loose comparison evaluation when checking if _offset_ is _null_, which means we are unable to set values for items at offset _0_.

## Steps to Reproduce

```
        $arrayList = new ArrayList(['change me', 'leave me']);
        $arrayList->count(); // 2
        $arrayList->offsetSet(0, 'new value');
        $arrayList->count(); // 3
        $arrayList->offsetGet(0); // 'change me', expected 'new value'
```

**Comment by sminnee** [comment]
Nice find - do you want to submit a fix as a PR to the 4.7 branch? Seems like a simple update of the relevant check to === null should be fine, but if any tests fail as a result it might need more discussion.

**Comment by emteknetnz** [comment]
Fixed in https://github.com/silverstripe/silverstripe-framework/pull/9800
